Uncaught error in Python test causes full riak_test run to fail

### What Happened

The python test has an error that causes the entire run of riak_test to halt.  
### What should have happened

Any error should simply be reported and riak_test should be able to continue.
### Test data

python test in riak_test branch `riak/1.3`

```
22:53:59.442 [info] rtdev:update_app_config(all, [{riak_kv,[{storage_backend,riak_kv_eleveldb_backend}]}])
22:53:59.443 [info] rtdev:update_app_config_file(/home/buildbot/test/rtee/current/dev/dev1/etc/app.config, [{riak_kv,[{storage_backend,riak_kv_eleveldb_backend}]}])
22:53:59.450 [info] rtdev:update_app_config_file(/home/buildbot/test/rtee/current/dev/dev2/etc/app.config, [{riak_kv,[{storage_backend,riak_kv_eleveldb_backend}]}])
22:53:59.458 [info] rtdev:update_app_config_file(/home/buildbot/test/rtee/current/dev/dev3/etc/app.config, [{riak_kv,[{storage_backend,riak_kv_eleveldb_backend}]}])
22:53:59.465 [info] rtdev:update_app_config_file(/home/buildbot/test/rtee/current/dev/dev4/etc/app.config, [{riak_kv,[{storage_backend,riak_kv_eleveldb_backend}]}])
22:53:59.472 [info] rtdev:update_app_config_file(/home/buildbot/test/rtee/current/dev/dev5/etc/app.config, [{riak_kv,[{storage_backend,riak_kv_eleveldb_backend}]}])
22:53:59.479 [info] rtdev:update_app_config_file(/home/buildbot/test/rtee/current/dev/dev6/etc/app.config, [{riak_kv,[{storage_backend,riak_kv_eleveldb_backend}]}])
22:53:59.487 [info] rtdev:update_app_config_file(/home/buildbot/test/rtee/riak_ee-1.2.1/dev/dev1/etc/app.config, [{riak_kv,[{storage_backend,riak_kv_eleveldb_backend}]}])
22:53:59.492 [info] rtdev:update_app_config_file(/home/buildbot/test/rtee/riak_ee-1.2.1/dev/dev2/etc/app.config, [{riak_kv,[{storage_backend,riak_kv_eleveldb_backend}]}])
22:53:59.498 [info] rtdev:update_app_config_file(/home/buildbot/test/rtee/riak_ee-1.2.1/dev/dev3/etc/app.config, [{riak_kv,[{storage_backend,riak_kv_eleveldb_backend}]}])
22:53:59.504 [info] rtdev:update_app_config_file(/home/buildbot/test/rtee/riak_ee-1.2.1/dev/dev4/etc/app.config, [{riak_kv,[{storage_backend,riak_kv_eleveldb_backend}]}])
22:53:59.510 [info] rtdev:update_app_config_file(/home/buildbot/test/rtee/riak_ee-1.2.1/dev/dev5/etc/app.config, [{riak_kv,[{storage_backend,riak_kv_eleveldb_backend}]}])
22:53:59.515 [info] rtdev:update_app_config_file(/home/buildbot/test/rtee/riak_ee-1.2.1/dev/dev6/etc/app.config, [{riak_kv,[{storage_backend,riak_kv_eleveldb_backend}]}])
22:53:59.521 [info] rtdev:update_app_config_file(/home/buildbot/test/rtee/riak_ee-1.3.2/dev/dev1/etc/app.config, [{riak_kv,[{storage_backend,riak_kv_eleveldb_backend}]}])
22:53:59.529 [info] rtdev:update_app_config_file(/home/buildbot/test/rtee/riak_ee-1.3.2/dev/dev2/etc/app.config, [{riak_kv,[{storage_backend,riak_kv_eleveldb_backend}]}])
22:53:59.536 [info] rtdev:update_app_config_file(/home/buildbot/test/rtee/riak_ee-1.3.2/dev/dev3/etc/app.config, [{riak_kv,[{storage_backend,riak_kv_eleveldb_backend}]}])
22:53:59.543 [info] rtdev:update_app_config_file(/home/buildbot/test/rtee/riak_ee-1.3.2/dev/dev4/etc/app.config, [{riak_kv,[{storage_backend,riak_kv_eleveldb_backend}]}])
22:53:59.550 [info] rtdev:update_app_config_file(/home/buildbot/test/rtee/riak_ee-1.3.2/dev/dev5/etc/app.config, [{riak_kv,[{storage_backend,riak_kv_eleveldb_backend}]}])
22:53:59.557 [info] rtdev:update_app_config_file(/home/buildbot/test/rtee/riak_ee-1.3.2/dev/dev6/etc/app.config, [{riak_kv,[{storage_backend,riak_kv_eleveldb_backend}]}])
22:53:59.601 [info] Checking for presence of python
22:53:59.614 [info] Checking for presence of virtualenv
22:53:59.626 [info] client_python_verify prereqs: [{"python",true},{"virtualenv",true}]
22:53:59.627 [notice] Running Test client_python_verify
22:54:01.630 [info] Test Runner `uname -a` : Linux riak-test-centos-6 2.6.32-358.6.2.el6.x86_64 #1 SMP Thu May 16 20:59:36 UTC 2013 x86_64 x86_64 x86_64 GNU/Linux

22:54:01.631 [info] [PREREQ] Cleaning up python scratch directory
22:54:01.643 [info] [PREREQ] Checking for python version >= 2.6, < 3.0
22:54:01.655 [info] [PREREQ] Got python version: Python 2.6.6

22:54:01.656 [info] [PREREQ] Checking for presence of setuptools
22:54:01.866 [info] [PREREQ] Cloning riak-python-client from git://github.com/basho/riak-python-client.git
22:54:02.368 [info] 
22:54:02.368 [info] 
22:54:02.369 [info] Initialized empty Git repository in /tmp/riak_test_scratch/riak-python-client/.git/
22:54:05.847 [info] [PREREQ] Resetting python client to tag '1.5-stable'
22:54:05.869 [info] 
22:54:05.869 [info] HEAD is now at 3f0c0c8 Merge pull request #273 from xen0n/update-bucket-search
22:54:05.887 [info] 
22:54:05.887 [info] Switched to a new branch '1.5-stable'
22:54:05.888 [info] [PREREQ] Installing an isolated environment with virtualenv in /tmp/riak_test_scratch/riak-python-client
22:54:06.342 [info] 
22:54:06.343 [info] 
22:54:06.343 [info] Not deleting ./bin
22:54:06.343 [info] New python executable in ./bin/python
22:54:06.344 [info] Installing setuptools...
22:54:06.545 [info] 
22:54:06.558 [info] .........
22:54:06.559 [info] done.
22:54:07.051 [info] Installing pip........
22:54:07.066 [info] .......
22:54:07.066 [info] done.
22:54:07.076 [info] Riak path: "/home/buildbot/test/rtee"
22:54:07.086 [info] rtdev:update_app_config_file(/home/buildbot/test/rtee/current/dev/dev1/etc/app.config, [{riak_search,[{enabled,true}]}])
22:54:07.096 [info] Running: /home/buildbot/test/rtee/current/dev/dev1/bin/riak start
22:54:12.154 [info] Wait until 'dev1@127.0.0.1' is pingable
22:54:12.306 [info] Wait until 'dev1@127.0.0.1' is pingable
22:54:12.307 [info] Check 'dev1@127.0.0.1' is a singleton
22:54:12.358 [info] Deployed nodes: ['dev1@127.0.0.1']
22:54:12.358 [info] Waiting for services [riak_kv] to start on ['dev1@127.0.0.1'].
22:54:18.367 [info] Connection Info: http: "127.0.0.1":10018 pb: "127.0.0.1":10017
22:54:18.368 [info] Enabling search hook on 'searchbucket'
22:54:18.368 [info] Installing search hook for bucket <<"searchbucket">>
22:54:19.005 [info] 
22:54:19.006 [info] 
22:54:19.006 [info] running develop
22:54:19.096 [info] 
22:54:19.096 [info] 
22:54:19.096 [info] running egg_info
22:54:19.097 [info] 
22:54:19.097 [info] 
22:54:19.097 [info] creating riak.egg-info
22:54:19.098 [info] writing requirements to riak.egg-info/requires.txt
22:54:19.098 [info] writing riak.egg-info/PKG-INFO
22:54:19.098 [info] 
22:54:19.099 [info] 
22:54:19.099 [info] writing top-level names to riak.egg-info/top_level.txt
22:54:19.099 [info] 
22:54:19.100 [info] 
22:54:19.100 [info] writing dependency_links to riak.egg-info/dependency_links.txt
22:54:19.102 [info] 
22:54:19.102 [info] 
22:54:19.102 [info] writing manifest file 'riak.egg-info/SOURCES.txt'
22:54:19.118 [info] 
22:54:19.118 [info] 
22:54:19.118 [info] reading manifest file 'riak.egg-info/SOURCES.txt'
22:54:19.119 [info] 
22:54:19.120 [info] 
22:54:19.120 [info] reading manifest template 'MANIFEST.in'
22:54:19.127 [info] 
22:54:19.128 [info] 
22:54:19.128 [info] writing manifest file 'riak.egg-info/SOURCES.txt'
22:54:19.133 [info] 
22:54:19.133 [info] 
22:54:19.133 [info] running build_ext
22:54:19.134 [info] 
22:54:19.134 [info] 
22:54:19.135 [info] Creating /tmp/riak_test_scratch/riak-python-client/lib/python2.6/site-packages/riak.egg-link (link to .)
22:54:19.135 [info] Adding riak 2.0.0.post4 to easy-install.pth file
22:54:19.136 [info] 
22:54:19.137 [info] 
22:54:19.137 [info] Installed /tmp/riak_test_scratch/riak-python-client
22:54:19.137 [info] 
22:54:19.138 [info] 
22:54:19.138 [info] Processing dependencies for riak==2.0.0.post4
22:54:19.139 [info] 
22:54:19.139 [info] 
22:54:19.139 [info] Searching for riak-pb>=1.4.0,<1.5.0
22:54:19.140 [info] 
22:54:19.140 [info] 
22:54:19.140 [info] Reading http://pypi.python.org/simple/riak_pb/
22:54:19.345 [info] 
22:54:19.346 [info] 
22:54:19.346 [info] Best match: riak-pb 1.4.1.1
22:54:19.346 [info] Downloading https://pypi.python.org/packages/2.6/r/riak_pb/riak_pb-1.4.1.1-py2.6.egg#md5=ee4d7cc7a0726520658728345aa2f41c
22:54:19.673 [info] 
22:54:19.673 [info] 
22:54:19.674 [info] Processing riak_pb-1.4.1.1-py2.6.egg
22:54:19.680 [info] 
22:54:19.680 [info] 
22:54:19.681 [info] Moving riak_pb-1.4.1.1-py2.6.egg to /tmp/riak_test_scratch/riak-python-client/lib/python2.6/site-packages
22:54:19.681 [info] 
22:54:19.681 [info] 
22:54:19.682 [info] Adding riak-pb 1.4.1.1 to easy-install.pth file
22:54:19.683 [info] 
22:54:19.683 [info] 
22:54:19.683 [info] Installed /tmp/riak_test_scratch/riak-python-client/lib/python2.6/site-packages/riak_pb-1.4.1.1-py2.6.egg
22:54:19.684 [info] 
22:54:19.684 [info] 
22:54:19.685 [info] Searching for protobuf==2.4.1
22:54:19.685 [info] 
22:54:19.685 [info] 
22:54:19.686 [info] Reading http://pypi.python.org/simple/protobuf/
22:54:19.844 [info] 
22:54:19.844 [info] 
22:54:19.844 [info] Link to http://code.google.com/p/protobuf/ ***BLOCKED*** by --allow-hosts
22:54:19.846 [info] 
22:54:19.846 [info] 
22:54:19.846 [info] Best match: protobuf 2.4.1
22:54:19.846 [info] 
22:54:19.846 [info] 
22:54:19.847 [info] Downloading https://pypi.python.org/packages/2.6/p/protobuf/protobuf-2.4.1-py2.6.egg#md5=5df1a5abfe927550a8643dd81c0442b7
22:54:20.648 [info] 
22:54:20.648 [info] 
22:54:20.648 [info] Processing protobuf-2.4.1-py2.6.egg
22:54:20.655 [info] 
22:54:20.655 [info] 
22:54:20.655 [info] Moving protobuf-2.4.1-py2.6.egg to /tmp/riak_test_scratch/riak-python-client/lib/python2.6/site-packages
22:54:20.656 [info] 
22:54:20.656 [info] 
22:54:20.656 [info] Adding protobuf 2.4.1 to easy-install.pth file
22:54:20.658 [info] 
22:54:20.658 [info] 
22:54:20.658 [info] Installed /tmp/riak_test_scratch/riak-python-client/lib/python2.6/site-packages/protobuf-2.4.1-py2.6.egg
22:54:20.659 [info] 
22:54:20.660 [info] 
22:54:20.660 [info] Searching for setuptools==0.6c11
22:54:20.660 [info] 
22:54:20.660 [info] 
22:54:20.660 [info] Best match: setuptools 0.6c11
22:54:20.660 [info] Processing setuptools-0.6c11-py2.6.egg
22:54:20.660 [info] setuptools 0.6c11 is already the active version in easy-install.pth
22:54:20.667 [info] 
22:54:20.667 [info] 
22:54:20.667 [info] Installing easy_install script to /tmp/riak_test_scratch/riak-python-client/bin
22:54:20.667 [info] 
22:54:20.667 [info] 
22:54:20.667 [info] Installing easy_install-2.6 script to /tmp/riak_test_scratch/riak-python-client/bin
22:54:20.667 [info] 
22:54:20.668 [info] 
22:54:20.668 [info] Using /tmp/riak_test_scratch/riak-python-client/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg
22:54:20.668 [info] 
22:54:20.668 [info] 
22:54:20.668 [info] Finished processing dependencies for riak==2.0.0.post4
22:54:20.668 [info] 
22:54:20.668 [info] 
22:54:20.668 [info] running test
22:54:20.718 [info] 
22:54:20.719 [info] 
22:54:20.719 [info] Searching for unittest2
22:54:20.719 [info] 
22:54:20.719 [info] 
22:54:20.719 [info] Reading http://pypi.python.org/simple/unittest2/
22:54:20.901 [info] 
22:54:20.901 [info] 
22:54:20.901 [info] Best match: unittest2 0.5.1
22:54:20.901 [info] 
22:54:20.901 [info] 
22:54:20.902 [info] Downloading https://pypi.python.org/packages/source/u/unittest2/unittest2-0.5.1.zip#md5=1527fb89e38343945af1166342d851ee
22:54:21.096 [info] 
22:54:21.096 [info] 
22:54:21.096 [info] Processing unittest2-0.5.1.zip
22:54:21.122 [info] 
22:54:21.123 [info] 
22:54:21.123 [info] Running unittest2-0.5.1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-0307E0/unittest2-0.5.1/egg-dist-tmp-MiPb_f
22:54:21.338 [info] 
22:54:21.339 [info] 
22:54:21.339 [info] zip_safe flag not set; analyzing archive contents...
22:54:21.339 [info] 
22:54:21.339 [info] 
22:54:21.339 [info] unittest2.collector: module references __file__
22:54:21.340 [info] 
22:54:21.340 [info] 
22:54:21.340 [info] unittest2.loader: module references __file__
22:54:21.343 [info] 
22:54:21.344 [info] 
22:54:21.344 [info] unittest2.test.test_discovery: module references __file__
22:54:21.510 [info] 
22:54:21.511 [info] 
22:54:21.511 [info] Installed /tmp/riak_test_scratch/riak-python-client/unittest2-0.5.1-py2.6.egg
22:54:21.515 [info] 
22:54:21.515 [info] 
22:54:21.516 [info] running build_ext
22:54:21.728 [info] 
22:54:21.729 [info] test_10 (riak.tests.test_feature_detection.FeatureDetectionTest) ... 
22:54:21.729 [info] ok
22:54:21.729 [info] test_11 (riak.tests.test_feature_detection.FeatureDetectionTest) ... ok
22:54:21.730 [info] test_12 (riak.tests.test_feature_detection.FeatureDetectionTest) ... ok
22:54:21.730 [info] test_12_loose (riak.tests.test_feature_detection.FeatureDetectionTest) ... ok
22:54:21.730 [info] test_14 (riak.tests.test_feature_detection.FeatureDetectionTest) ... ok
22:54:21.731 [info] test_implements_server_version (riak.tests.test_feature_detection.FeatureDetectionTest) ... ok
22:54:21.731 [info] test_pre_10 (riak.tests.test_feature_detection.FeatureDetectionTest) ... ok
22:54:21.731 [info] test_clear (riak.tests.test_pool.PoolTest) ... 
22:54:22.259 [info] 
22:54:22.260 [info] 
22:54:22.260 [info] ok
22:54:22.261 [info] 
22:54:22.261 [info] 
22:54:22.262 [info] test_filter_skips_unmatching_elements (riak.tests.test_pool.PoolTest) ... ok
22:54:22.262 [info] test_iteration (riak.tests.test_pool.PoolTest) ... 
22:54:22.804 [info] 
22:54:22.805 [info] 
22:54:22.805 [info] ERROR
22:54:22.805 [info] 
22:54:22.805 [info] 
22:54:22.806 [info] test_reentrance (riak.tests.test_pool.PoolTest) ... ok
22:54:22.806 [info] test_removes_bad_resource (riak.tests.test_pool.PoolTest) ... ok
22:54:22.806 [info] test_requires_filter_to_be_callable (riak.tests.test_pool.PoolTest) ... ok
22:54:22.806 [info] test_stress (riak.tests.test_pool.PoolTest) ... ERROR
22:54:22.807 [info] 
22:54:22.807 [info] 
22:54:22.807 [info] test_thread_safety (riak.tests.test_pool.PoolTest) ... ERROR
22:54:22.808 [info] test_unlocks_when_exception_raised (riak.tests.test_pool.PoolTest) ... ok
22:54:22.808 [info] test_yields_default_when_empty (riak.tests.test_pool.PoolTest) ... ok
22:54:22.808 [info] 
22:54:22.809 [info] 
22:54:22.809 [info] test_yields_new_object_when_empty (riak.tests.test_pool.PoolTest) ... ok
22:54:22.810 [info] test_yields_same_object_in_serial_access (riak.tests.test_pool.PoolTest) ... ok
22:54:22.810 [info] test_merge_riak_core_options (riak.tests.test_server_test.TestServerTestCase) ... ok
22:54:22.810 [info] test_merge_riak_kv_options (riak.tests.test_server_test.TestServerTestCase) ... ok
22:54:22.811 [info] test_merge_riak_search_options (riak.tests.test_server_test.TestServerTestCase) ... ok
22:54:22.811 [info] 
22:54:22.811 [info] 
22:54:22.811 [info] test_merge_vmargs (riak.tests.test_server_test.TestServerTestCase) ... ok
22:54:22.812 [info] test_options_defaults (riak.tests.test_server_test.TestServerTestCase) ... ok
22:54:22.812 [info] test_set_default_tmp_dir (riak.tests.test_server_test.TestServerTestCase) ... ok
22:54:22.812 [info] test_set_non_default_tmp_dir (riak.tests.test_server_test.TestServerTestCase) ... ok
22:54:22.813 [info] test_set_ring_state_dir (riak.tests.test_server_test.TestServerTestCase) ... ok
22:54:23.013 [info] 
22:54:23.014 [info] test_add (riak.tests.test_all.FilterTests) ... 
22:54:23.014 [info] ok
22:54:23.015 [info] test_and (riak.tests.test_all.FilterTests) ... ok
22:54:23.015 [info] test_chaining (riak.tests.test_all.FilterTests) ... ok
22:54:23.016 [info] 
22:54:23.016 [info] test_multi_and (riak.tests.test_all.FilterTests) ... ok
22:54:23.016 [info] test_multi_or (riak.tests.test_all.FilterTests) ... ok
22:54:23.017 [info] test_or (riak.tests.test_all.FilterTests) ... ok
22:54:23.017 [info] test_simple (riak.tests.test_all.FilterTests) ... ok
22:54:23.017 [info] test_add_document_to_index (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:23.018 [info] 
22:54:23.018 [info] /tmp/riak_test_scratch/riak-python-client/riak/util.py:148: UserWarning: ``solr`` is deprecated, use ``fulltext_search``, ``fulltext_add`` and ``fulltext_delete`` directly
22:54:23.019 [info]   value = self.fget(obj)
22:54:23.433 [info] 
22:54:23.434 [info] ok
22:54:23.508 [info] test_add_multiple_documents_to_iindex (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:23.509 [info] ok
22:54:23.509 [info] 
22:54:23.509 [info] test_bad_key (riak.tests.test_all.RiakHttpTransportTestCase) ... ok
22:54:23.510 [info] test_binary_store_and_get (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:23.613 [info] 
22:54:23.613 [info] ok
22:54:23.613 [info] test_blank_binary_204 (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:23.631 [info] 
22:54:23.631 [info] ok
22:54:23.733 [info] test_bucket_delete (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:23.733 [info] ok
22:54:23.744 [info] test_bucket_search_enabled (riak.tests.test_all.RiakHttpTransportTestCase) ... ok
22:54:23.745 [info] test_clear_bucket_properties (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:24.159 [info] 
22:54:24.159 [info] 
22:54:24.159 [info] ok
22:54:24.160 [info] 
22:54:24.297 [info] test_client_exceptional_paths (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:24.298 [info] ok
22:54:24.298 [info] test_counter_ops (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:24.312 [info] 
22:54:24.312 [info] ERROR
22:54:24.325 [info] test_counter_requires_allow_mult (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:24.326 [info] ok
22:54:24.326 [info] test_custom_bucket_encoder_decoder (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:24.376 [info] 
22:54:24.377 [info] ok
22:54:24.377 [info] test_custom_client_encoder_decoder (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:24.427 [info] 
22:54:24.427 [info] ok
22:54:24.495 [info] test_delete (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:24.495 [info] ok
22:54:24.605 [info] test_delete_documents_from_search_by_id (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:24.605 [info] ok
22:54:24.605 [info] test_delete_documents_from_search_by_query (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:24.719 [info] 
22:54:24.719 [info] ok
22:54:24.832 [info] test_delete_documents_from_search_by_query_and_id (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:24.832 [info] ok
22:54:25.618 [info] test_disable_search_commit_hook (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:25.618 [info] 
22:54:25.618 [info] ok
22:54:25.619 [info] 
22:54:25.993 [info] test_enable_search_commit_hook (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:25.994 [info] ok
22:54:25.994 [info] test_erlang_map_reduce (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:26.285 [info] 
22:54:26.285 [info] 
22:54:26.286 [info] ok
22:54:26.286 [info] 
22:54:26.604 [info] test_erlang_source_map_reduce (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:26.604 [info] ok
22:54:26.764 [info] test_filter_not_found (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:26.764 [info] ok
22:54:26.891 [info] test_generate_key (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:26.891 [info] ok
22:54:26.968 [info] test_if_none_match (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:26.968 [info] ok
22:54:27.194 [info] test_index_eq_query_return_terms (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:27.194 [info] 
22:54:27.194 [info] ok
22:54:27.195 [info] 
22:54:27.394 [info] test_index_eq_query_stream_return_terms (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:27.395 [info] ERROR
22:54:27.591 [info] test_index_pagination (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:27.591 [info] FAIL
22:54:27.820 [info] test_index_pagination_return_terms (riak.tests.test_all.RiakHttpTransportTestCase) ... FAIL
22:54:27.820 [info] test_index_pagination_stream (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:28.014 [info] 
22:54:28.015 [info] 
22:54:28.015 [info] ERROR
22:54:28.015 [info] 
22:54:28.208 [info] test_index_pagination_stream_return_terms (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:28.208 [info] ERROR
22:54:28.395 [info] test_index_return_terms (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:28.395 [info] FAIL
22:54:28.409 [info] test_is_alive (riak.tests.test_all.RiakHttpTransportTestCase) ... ok
22:54:28.409 [info] test_javascript_arg_map_reduce (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:28.518 [info] ok
22:54:28.518 [info] test_javascript_bucket_map_reduce (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:28.787 [info] 
22:54:28.787 [info] ok
22:54:28.879 [info] test_javascript_named_map (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:28.879 [info] ok
22:54:29.064 [info] test_javascript_named_map_reduce (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:29.064 [info] 
22:54:29.064 [info] ok
22:54:29.065 [info] 
22:54:29.157 [info] test_javascript_source_map (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:29.158 [info] ok
22:54:29.342 [info] test_javascript_source_map_reduce (riak.tests.test_all.RiakHttpTransportTestCase) ... ok
22:54:29.547 [info] test_key_filters (riak.tests.test_all.RiakHttpTransportTestCase) ... ok
22:54:29.733 [info] test_key_filters_f_chain (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:29.733 [info] ok
22:54:29.733 [info] 
22:54:29.733 [info] test_key_filters_with_search_query (riak.tests.test_all.RiakHttpTransportTestCase) ... ok
22:54:29.734 [info] test_link_walking (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:30.049 [info] 
22:54:30.049 [info] 
22:54:30.049 [info] ok
22:54:30.050 [info] 
22:54:30.124 [info] test_list_buckets (riak.tests.test_all.RiakHttpTransportTestCase) ... ok
22:54:30.239 [info] test_map_reduce_from_object (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:30.240 [info] ok
22:54:30.394 [info] test_map_values (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:30.394 [info] ok
22:54:30.552 [info] test_map_values_json (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:30.553 [info] ok
22:54:30.562 [info] test_missing_object (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:30.563 [info] ok
22:54:31.491 [info] test_mr_list_add (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:31.491 [info] 
22:54:31.492 [info] ok
22:54:31.492 [info] 
22:54:32.419 [info] test_mr_list_add_mix (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:32.420 [info] 
22:54:32.420 [info] ok
22:54:32.420 [info] 
22:54:33.393 [info] test_mr_list_add_two_buckets (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:33.393 [info] 
22:54:33.393 [info] ok
22:54:33.393 [info] 
22:54:33.535 [info] test_multiget_bucket (riak.tests.test_all.RiakHttpTransportTestCase) ... 
22:54:33.536 [info] ERROR
22:54:33.536 [info] test_multiget_errors (riak.tests.test_all.RiakHttpTransportTestCase) ... 
23:04:33.539 [warning] client_python_verify failed: {{assertEqual_failed,[{module,client_python_verify},{line,38},{expression,"ExitCode"},{expected,0},{value,-1}]},[{client_python_verify,'-confirm/0-fun-0-',2,[{file,"tests/client_python_verify.erl"},{line,38}]},{client_python_verify,confirm,0,[{file,"tests/client_python_verify.erl"},{line,38}]}]}
23:04:33.539 [error] Error in process <0.328.0> on node 'riak_test@127.0.0.1' with exit value: {{assertEqual_failed,[{module,client_python_verify},{line,38},{expression,"ExitCode"},{expected,0},{value,-1}]},[{client_python_verify,'-confirm/0-fun-0-',2,[{file,"tests/client_python_verify.erl"},{line,38}]},{client_python_verify... 


23:04:33.541 [error] 
================ client_python_verify failure stack trace =====================
{{assertEqual_failed,[{module,client_python_verify},
                      {line,38},
                      {expression,"ExitCode"},
                      {expected,0},
                      {value,-1}]},
 [{client_python_verify,'-confirm/0-fun-0-',2,
                        [{file,"tests/client_python_verify.erl"},{line,38}]},
  {client_python_verify,confirm,0,
                        [{file,"tests/client_python_verify.erl"},{line,38}]}]}
===============================================================================

23:04:33.541 [notice] client_python_verify Test Run Complete
escript: exception exit: {eagain,[{erlang,open_port,
                                 [{spawn,"/bin/sh -s unix:cmd 2>&1"},[stream]],
                                 []},
                         {os,start_port_srv_handle,1,
                             [{file,"os.erl"},{line,254}]},
                         {os,start_port_srv_loop,0,
                             [{file,"os.erl"},{line,270}]}]}
  in function  os:unix_cmd/1 (os.erl, line 197)
  in call from rtdev:'-stop_all/1-fun-0-'/1 (src/rtdev.erl, line 276)
  in call from rtdev:'-stop_all/1-lc$^0/1-0-'/2 (src/rtdev.erl, line 283)
  in call from rtdev:stop_all/1 (src/rtdev.erl, line 283)
  in call from rtdev:'-teardown/0-lc$^0/1-0-'/1 (src/rtdev.erl, line 468)
  in call from riak_test:run_test/6 (src/riak_test.erl, line 240)
  in call from riak_test:'-main/1-lc$^1/1-1-'/5 (src/riak_test.erl, line 138)
  in call from riak_test:'-main/1-lc$^1/1-1-'/5 (src/riak_test.erl, line 138)
make: *** [test-ee] Error 127
program finished with exit code 2
```

--- a/tests/client_python_verify.erl
+++ b/tests/client_python_verify.erl
@@ -14,7 +14,7 @@
 -prereq("virtualenv").
 
 confirm() ->
-    prereqs(),
+    {ok, TestCommand} = prereqs(),
     Config = [{riak_search, [{enabled, true}]}],
     [Node] = rt:deploy_nodes(1, Config),
     rt:wait_for_service(Node, riak_search),
@@ -28,7 +28,7 @@ confirm() ->
     lager:info("Enabling search hook on 'searchbucket'"),
     rt:enable_search_hook(Node, <<"searchbucket">>),
 
-    {ExitCode, PythonLog} = rt_local:stream_cmd("bin/python setup.py develop test",
+    {ExitCode, PythonLog} = rt_local:stream_cmd(TestCommand,
                                           [{cd, ?PYTHON_CHECKOUT},
                                            {env,[{"RIAK_TEST_PB_HOST", PB_Host},
                                                  {"RIAK_TEST_PB_PORT", integer_to_list(PB_Port)},
@@ -72,4 +72,14 @@ prereqs() ->
 
     lager:info("[PREREQ] Installing an isolated environment with virtualenv in ~s", [?PYTHON_CHECKOUT]),
     rt_local:stream_cmd("virtualenv --clear --no-site-packages .", [{cd, ?PYTHON_CHECKOUT}]),
-    ok.
+
+    lager:info("[PREREQ] Installing dependencies"),
+    rt_local:stream_cmd("bin/python setup.py develop", [{cd, ?PYTHON_CHECKOUT}]),
+    case Minor of
+        $6 ->
+            lager:info("[PREREQ] Installing unittest2 for python 2.6"),
+            rt_local:stream_cmd("bin/easy_install unittest2", [{cd, ?PYTHON_CHECKOUT}]),
+            {ok, "bin/unit2 riak.tests.test_all"};
+        _ ->
+            {ok, "bin/python setup.py test"}
+    end.

